tb_plugin/torch_tb_profiler/profiler/op_tree.py

```diff
@@ -0,0 +1,329 @@
+import sys
+from collections import defaultdict
+
+from .. import utils
+from .node import (BackwardNode, ModuleNode, OperatorNode, ProfilerStepNode,
+                   RuntimeNode, is_operator_node)
+from .trace import EventTypes
+
+logger = utils.get_logger()
+
+
+class OpTreeBuilder:
+    BACKWARD_ROOT_PREFIX = 'autograd::engine::evaluate_function:'
+    BACKWARD_ACCUMULATE_GRAD = 'autograd::engine::evaluate_function: torch::autograd::AccumulateGrad'
+
+    def __init__(self):
+        self.main_tid = None
+        self.tid2tree = None
+
+    def build_tree(self, tid2list, tid2zero_rt_list, staled_device_nodes, fwd_bwd_map):
+        '''Construct the BackwardNode and replace the original backward nodes
+        '''
+        self.tid2tree = self._build_tree(tid2list, tid2zero_rt_list, staled_device_nodes)
+        self._set_main_tid()
+
+        modules, backward_nodes = self._get_modules()
+        if not modules or not backward_nodes:
```

Comment:
does module view support inference scenario? there are modules but not backward_nodes?